### ISSUE TITLE
Problems with built-on Audio output since Sonoma 14.4

### ISSUE THREAD

**Issue by gsternagl** [zearp/Nucintosh#155]
- I upgraded to Sonoma 14.4 from Ventura 13.6.6
- I am on OC 0.99 and the latest EFI folder from you including latest Lilu .kextand AppleALC.kext (layout-id = 14)
- No issues overall until I played some sound
- Audio output works for ~30s and then goes away (becomes so quiet that you can barely hear it anymore)
- When opening the MacOS Audio Settings panel and switch to another Audio Input, Output sound comes back again
- Did not change the HW in any way and did not have any problems with Audio on Ventura or before
- I have a USB webcam with a builtin Microphone that I primarily use for conf-calls. But that was there in Ventura and before
- I checked with IOregistryexplorer whether AppleALC patches AppleHDA correctly and everything seems OK
- Downgrading to 13.6.6 agains since this is my main machine

**Comment by zearp** [comment]
I assume you're talking about the line-out/headphone jack. I do not use that myself as I use audio over usb to my amp but as a test I've connected the the headphone output to the line input of my amp and been playing some Youtube playlist for the past hour. The audio works fine, no drop outs or artefacts.

I don't know how you use OpenCore 0.9.9. + stuff from the repo. The repo is on 1.0.0 and should be used as is. Not mixing in other kexts even if they are newer. It shouldn't cause any issues but for me in order to be able to troubleshoot we both need to run the exact same EFI with just changes to the config (serials etc).

I will keep this connected for now and see if anything changes. Personally I prefer using audio over usb and my amp can do both but headphone/line-out should work as well. So this a good test.

As a test can you delete your whole EFI and download the latest release and only copy over your serials/network mac address and such to the new config and try again? That way we'll both run the the same version of OpenCore and all accompanying kexts. 

For now I attach the can not reproduce label as I really can't reproduce it. Which always sucks to solve issues people have. I need to be able to reproduce it else all I can do is guess.

**Comment by zearp** [comment]
I assume you're talking about the line-out/headphone jack. I do not use that myself as I use audio over usb to my amp but as a test I've connected the the headphone output to the line input of my amp and been playing some Youtube playlist for the past hour. The audio works fine, no drop outs or artefacts.

I don't know how you use OpenCore 0.9.9 + stuff from the repo. The repo is on 1.0.0 and should be used as is. Not mixing in other kexts even if they are newer. It shouldn't cause any issues but for me in order to be able to troubleshoot we both need to run the exact same EFI with just changes to the config (serials etc).

I will keep this connected for now and see if anything changes. Personally I prefer using audio over usb and my amp can do both but headphone/line-out should work as well. So this a good test.

As a test can you delete your whole EFI and download the latest release and only copy over your serials/network mac address and such to the new config and try again? That way we'll both run the the same version of OpenCore and all accompanying kexts. 

For now I attach the can not reproduce label as I really can't reproduce it. Which always sucks to solve issues people have. I need to be able to reproduce it else all I can do is guess.

**Comment by gsternagl** [comment]
Thanks for trying to reproduce. Yes I am talking about the headphone jack at the front. I've just finished doing a restore of Ventura and Audio works again as expected. I did make some minor adjustments to my EFI-config.plist. I still have a fresh Sonoma installation on an external SSD and I can run it with an 1:1 copy of your EFI on an USB stick so that we have the exact same baseline. Will test that later tonight and get back to you.

**Comment by zearp** [comment]
You're welcome! I always do my best and as long as I can reproduce usually can fix it or come up with some work around. I am running 14.5. beta 3 on my "vanilla" test NUC but that shouldn't matter much.

(My other NUCs are still on Ventura with Broadcom and kind of stuck there as I want the iServices and full disk encryption to work. While full disk encryption works in Sonoma with Intel wireless the iServices integration is almost non-existent. Still surprised Apple actually removed Broadcom support from Sonoma completely as it stops many old but still capable Macs from upgrading to Sonoma unless you resort to OCLP which has as downsides you can no longer use full disk encryption.)

**Comment by zearp** [comment]
You're welcome! I always do my best and as long as I can reproduce usually can fix it or come up with some work around. I am running 14.5. beta 3 on my "vanilla" test NUC but that shouldn't matter much.

(My other NUCs are still on Ventura with Broadcom and kind of stuck there as I want the iServices and full disk encryption to work. While full disk encryption works in Sonoma with Intel wireless the iServices integration is almost non-existent. Still surprised Apple actually removed Broadcom support from Sonoma completely as it stops many old but still capable Macs from upgrading to Sonoma unless you resort to OCLP which has as downside you can no longer use full disk encryption.)

**Comment by gsternagl** [comment]
Apple wants to transition their Intel customerbase to Apple Silicon asap and as a "nice" side-effect they will kill our community unfortunately. I have a mix of Apple and Hackintoshes but I don't like with their new systems that you cannot change or expand anything in them. It's all soldered on that MB and if you want more RAM, GFX, etc. you pay a fortune. 

Anyways back to the technical issue: I did boot my fresh Sonoma installation with your original EFI release from end of March. The only thing I changed was the serial, ROM, MLB and System UUID. Now interestingly the same issue occured but in a different way. 
- I start a browser with YT and play some video, sound works
- Then I start in another browser a Google Meet session, just to make use of the microphone
- After exactly 30s the output sound of YT doesn't go away anymore but it sounds like it's being switched from Stereo to Mono. It sounds noticable different if you have headphones on.
- It differs to my previous issue in that with my own original EFI (based on an older version of yours and upgraded to OC 0.99), the sound goes away nearly completely, also after 30 secs. 
- When I then go back to the System Settings -> Audio settings and change the Input from one to another e.g. from "Line-In" to "Internal Microphone", Output Audio will go back to normal (Stereo) for ~30 secs and then again go back into Mono.
- That 30 sec interval is pretty consistent
- I also disconnected my USB webcam with the internal microphone but that didn't change the issue

I will be staying on Ventura for now as this works fine. But thought this might be interesting for other people to know who are considering moving to Sonoma. Of course only if somebody can reproduce this issue.

**Comment by zearp** [comment]
Fully agreed!

As for the issue, I will try and think of what I can use here to use the mic in one app and also have audio coming from other apps. I do share movies on Discord from time to time where I need to use an aggregate device but that does work with audio coming from both the video player and Discord plus discord also using the microphone. 

https://github.com/ExistentialAudio/BlackHole

I wonder if it's a privacy thing or something like that? You could check in the Console app if there's any clues, it always happens after 30 seconds that might help in spotting it. It can display (filtered) logs in real time too. I don't have Google Meet but I will try and think of something else than Discord to test. We're talking Safari here?

**Comment by gsternagl** [comment]
Yes that's a good point (Console). I tried to check registers for the Audio device via IOregistryexplorer and also dumped the content of with alc-dump (part of AppleALC source) but didn't think of the obvious and simple. There was nothing obvious when the Audio went off. So it doesn't seem to be any registers. It seems that when the System Settings Audio panel for Inputs is open, Output Audio will stay on for a longer period.

For reproducing: I believe you just need any application that will continuously use the built-in or Line-In microphone for Audio Input. Facetime should work as well. But if you have a google email address (also required for YT) then just open any webbrowser with https://meet.google.com and login with your normal google account. Then start a new video conference (with yourself).

I would have loved to debug the AppleALC.kext but that seems to be a more complex effort and since Ventura still works I will opt for that for now.

Thanks for trying

**Comment by zearp** [comment]
When starting a meet with myself I can see the orange dot and some indication of audio behind my profile pic when there's audio picked up by the mic. Youtube kept playing fine on another tab but the audio quality degraded a bit. When forcing the headphone jack to use the max bitrate/freq the weird sounds (some kind artefacts) went away. Maybe this will help you too. you can change it in the ```Audio MIDI Setup``` app.

![Screenshot 2024-04-25 at 20 44 44](https://github.com/zearp/Nucintosh/assets/63272687/8a4b803c-18f8-4fa6-91da-ecb499ff9411)

![Screenshot 2024-04-25 at 20 44 34](https://github.com/zearp/Nucintosh/assets/63272687/556d5b2a-a007-4d49-984e-9c8f6c7afac2)

Changing the alc layout could also be an option but I am not sure and there [many layouts](https://github.com/acidanthera/AppleALC/blob/master/Resources/ALC235/Info.plist) to try for our chip (235). Making a custom one just for this NUC is something I haven't gotten in to so I don't know if that's easy or hard. And it may be a waste of time if the layout has nothing to do with your audio dropping out or degrades in quality.

Adding the Debug Enhancer kext will give a lot more logs without having to into debug kexts and verbosity levels right away, which may generate a lot of logs.

**Comment by gsternagl** [comment]
Ah at least you were able to reproduce. Good. I forgot to mention that I also already tried to change sample rates via Audio MIDI. It reacts similar as Audio / Input panel in that when you change the Input Settings or any of the Sample Rates, Output Audio goes back to normal for ~30 sec and then back to strange state. 
I did have a look at console output and compared with my Ventura setup. There is nothing obvious.
**coreaudiod** reports deviated sample times (but only marginal) every minute or so and then I have a lot of error messages from **runningboardd** about a memorystatus_control_error. But they occur on both Ventura and Sonoma and don't seem to be related to the cause of the issue. The runninboardd messages occur a lot. I looked up on Acidantera and really RestrictEvents.kext should help to reduce these. But it doesn't seem to as I have it installed already. CustomMemory configuration could help, even though my Memory modules show up correctly in System Details, including Vendor ID, Tye, Serial Nr. etc. 
But as I said I don't think it's related to the audio problem.
